enterprise/cmd/frontend/internal/repos/webhooks/bitbucket_server_handler.go

```diff
@@ -0,0 +1,77 @@
+package webhooks
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	"github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/webhooks"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/errcode"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/bitbucketserver"
+	"github.com/sourcegraph/sourcegraph/internal/repoupdater"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type BitbucketServerHandler struct {
+	logger log.Logger
+}
+
+func (g *BitbucketServerHandler) Register(router *webhooks.Router) {
+	router.Register(func(ctx context.Context, _ database.DB, _ extsvc.CodeHostBaseURL, payload any) error {
+		return g.handlePushEvent(ctx, payload)
+	}, extsvc.KindBitbucketServer, "repo:refs_changed")
+}
+
+func NewBitbucketServerHandler() *BitbucketServerHandler {
+	return &BitbucketServerHandler{
+		logger: log.Scoped("webhooks.BitbucketServerHandler", "bitbucket server webhook handler"),
+	}
+}
+
+func (g *BitbucketServerHandler) handlePushEvent(ctx context.Context, payload any) error {
+	event, ok := payload.(*bitbucketserver.PushEvent)
+	if !ok {
+		return errors.Newf("expected BitbucketServer.PushEvent, got %T", payload)
+	}
+
+	repoName, err := bitbucketServerNameFromEvent(event)
+	if err != nil {
+		return errors.Wrap(err, "handleGitLabWebhook: get name failed")
+	}
+
+	resp, err := repoupdater.DefaultClient.EnqueueRepoUpdate(ctx, repoName)
+	if err != nil {
+		// Repo not existing on Sourcegraph is fine
+		if errcode.IsNotFound(err) {
+			g.logger.Warn("BitbucketServer push webhook received for unknown repo", log.String("repo", string(repoName)))
+			return nil
+		}
+		return errors.Wrap(err, "handlePushEvent: EnqueueRepoUpdate failed")
+	}
+
+	g.logger.Info("successfully updated", log.String("name", resp.Name))
+	return nil
+}
+
+func bitbucketServerNameFromEvent(event *bitbucketserver.PushEvent) (api.RepoName, error) {
+	if event == nil {
+		return "", errors.New("URL for repository not found")
```

Comment:
Why `URL for repository`? Should this be something like "no event found"?

Suggestion in case you want to accept it: 

```suggestion
		return "", errors.New("no event found")
```

Reply:
I made it more expicit: `nil PushEvent received`